clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -4077,13 +4077,9 @@ void Sema::CheckSYCLKernelCall(FunctionDecl *KernelFunc,
 
   // check that calling kernel conforms to spec
   QualType KernelParamTy = KernelFunc->getParamDecl(0)->getType();
-  if (KernelParamTy->isReferenceType()) {
-    // passing by reference, so emit warning if not using SYCL 2020
-    if (LangOpts.getSYCLVersion() < LangOptions::SYCL_2020)
-      Diag(KernelFunc->getLocation(), diag::warn_sycl_pass_by_reference_future);
-  } else {
+  if (!(KernelParamTy->isReferenceType())) {
     // passing by value.  emit warning if using SYCL 2020 or greater
-    if (LangOpts.getSYCLVersion() > LangOptions::SYCL_2017)
+    if (LangOpts.getSYCLVersion() >= LangOptions::SYCL_2020)
```

Comment:
Wonder if its time to just emit it as an error, or completely stop diagnosing

Reply:
Removed the unnecessary check about SYCL version (we'll always be on SYCL 2020+ after this removal) but keeping the warning for this one.